### PR TITLE
SYNCT-65: Added possibility to sync the privilege object

### PR DESCRIPTION
**Pull request by areklalo**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java [hunk 1]
```diff
@@ -62,35 +68,57 @@ private Object retrieveRestObject(String category, String uuid) {
...
+                service.voidPatient(retrievedPatient, VOIDING_REASON);
+                break;
+            case ACTION_UPDATED:
+            default:
+                Context.getPatientService().savePatient(patient);
+                break;
+        }
+    }
+
+    private void persistOpenMrsPrivelege(Privilege privilege, String action) {
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java]: Typo

File: api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java [hunk 2]
```diff
@@ -62,35 +68,57 @@ private Object retrieveRestObject(String category, String uuid) {
...
-                    service.voidPatient(patient, VOIDING_REASON);
-                    break;
-                case ACTION_UPDATED:
-                default:
-                    Context.getPatientService().savePatient((Patient)object);
-                    break;
-            }
+            persistOpenMrsPatient((Patient) object, action);
+        } else if (object instanceof Privilege) {
+            persistOpenMrsPrivelege((Privilege) object, action);
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java]: Typo

File: api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java (hunk 1 above)
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java]: Typo

File: api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java (hunk 2 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java]: Thanks, corrected.

File: api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java (hunk 1 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/sync2/api/sync/SyncPersistence.java]: Thanks, corrected.

**Comment by pkornowski** [comment]
When I go to the Audit Log page and filter by privilege only there is 500 from the server. You need to add Resource enum PRIVILEGE for fix this.

**Pull request closed without merging**